clash_royale_service.py

```diff
@@ -257,6 +257,43 @@ def clan_remaining_war_attacks(self, clan_tag: str) -> List[ClanRemainingWarAtta
                     ),
                 )
             )
-
         all_clan_attacks.sort(key=lambda x: (x.medals, x.fame), reverse=True)
         return all_clan_attacks
+
+    def clan_players_remaining_war_attacks(self, clan_tag: str) -> List[PlayersRemainingWarAttacks]:
+        curr_river_race = self.clash_royale_client.get_current_river_race(
+            clan_tag)
+        curr_river_race_participants = {}
+        all_current_war_players = {}
+        all_current_war_players_output = []
+        for participant in curr_river_race['clan']['participants']:
+            curr_river_race_participants[participant['tag']] = participant['decksUsedToday']
+            # Captures war participants who may not currently be in clan but attempted at least one attack
+            if participant['decksUsedToday'] > 0 and participant['decksUsedToday'] < 4:
+                all_current_war_players[participant['tag']] = {
+                    'name': participant['name'],
+                    'decksUsedToday': participant['decksUsedToday'],
+                    'lastSeen': None,
+                    'inClan': False
+                }
+        clan_info = self.clash_royale_client.get_clan_info(clan_tag)
+        # Captures everyone in the clan and checks their decksUsedToday        
+        for clan_member in clan_info['memberList']:
+            if clan_member['tag'] in curr_river_race_participants and curr_river_race_participants[clan_member['tag']] < 4:
+                all_current_war_players[clan_member['tag']] = {
+                    "name": clan_member['name'],
+                    'decksUsedToday': curr_river_race_participants[clan_member['tag']],
+                    'lastSeen': clan_member['lastSeen'],
+                    'inClan': True
+                }

```

Comment:
Given that the if condition already finds `clan_member['tag'] in curr_river_race_participants`, you know it exists with the values set in above for loop, why not only update the specific fields which have `different` values. i.e.
```python
all_current_war_players[clan_member['tag']]['lastSeen'] = clan_member['lastSeen']
all_current_war_players[clan_member['tag']]['inClan'] = True
```

Reply:
that's true, big brains.

Reply:
i did think about this but i got lazy. soz

Reply:
did a bit of digging, this is not necessarily true. A new member may join the clan without doing war, and they are excluded from the all_current_war_players list.

Reply:
You know, I'm looking at this more and it's making me wonder.... why aren't we just re-using the same / similar approach/structure/logic/flow as 
<img width="736" alt="image" src="https://user-images.githubusercontent.com/18656129/205911969-fb29c3cc-53ba-4a60-990c-25649bd2c8d1.png">

See how straightforward and simple `_get_players_remaining` is? Why can't we just do something like that instead of these 3 separate for loops? Cuz ngl, it was a bit confusing processing wtf was going on.

Action -> You can simplify this down to 1 for loop (which will be easier to process/read the code too). Please simplify, we can pair program if need be

Reply:
done